main.py

```diff
@@ -15,54 +15,125 @@
 device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 cwd = os.getcwd()
 
-with open(os.path.join(cwd,'config.json')) as f:
-	CONFIGURATION = json.load(f)
-	
+with open(os.path.join(cwd, 'config.json')) as f:
+    CONFIGURATION = json.load(f)
+
 hparams = CONFIGURATION['HPARAMS']
 hparams['device'] = device
 
+
 def main():
-	# Choosing image transformations
-	transform = transforms.Compose([
-		transforms.ToTensor(),
-		transforms.Resize((hparams['IMAGE_SIZE'],hparams['IMAGE_SIZE'])),
-		# The normalize parameters depends on the model we're gonna use
-		# If we apply transfer learning from a model that used ImageNet, then
-		# we should use the ImageNet values to normalize the dataset.
-		# Otherwise we could just normalize the values between -1 and 1 using the 
-		# standard mean and standard deviation
-		# transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225]),
-		transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5]),
-	])
-	dataset = Flickr8kDataset(transform=transform,reduce=True, vocab_max_size=hparams['VOCAB_SIZE'])
-
-	# Test the dataloader
-	model = ImageCaptioningModel(
-		embed_size=hparams['EMBED_SIZE'],
-		image_features_dim=512,
-		vocab_size=len(dataset.vocab.word_to_index),
-		attention_dim=hparams['ATTENTION_DIM'],
-		caption_max_length=hparams['MAX_LENGTH']
-	)
-
-	train_split, test_split = random_split(dataset,[32364,8091]) #80% train, 20% test
-	train_loader = DataLoader(train_split, shuffle=True, batch_size=hparams['BATCH_SIZE'], collate_fn=CapsCollate(pad_idx=dataset.vocab.word_to_index['<PAD>'],batch_first=True))
-	test_loader = DataLoader(test_split,batch_size=hparams['BATCH_SIZE'], collate_fn=CapsCollate(pad_idx=dataset.vocab.word_to_index['<PAD>'],batch_first=True))
-	
-	optimizer = optim.Adam(model.parameters(),lr=hparams['LEARNING_RATE'])
-	criterion = nn.CrossEntropyLoss(ignore_index=dataset.vocab.word_to_index['<PAD>'])
-
-	train(
-		num_epochs=hparams['NUM_EPOCHS'],
-		model=model,
-		train_loader=train_loader,
-		test_loader=test_loader,
-		optimizer= optimizer,
-		criterion=criterion,
-		device=hparams['device'],
-		log_interval=hparams['LOG_INTERVAL']
-	)
+    # Choosing image transformations
+    transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Resize(hparams['image_size']),
+        transforms.CenterCrop(hparams['image_size']),
```

Comment:
hparams['IMAGE_SIZE']

Reply:
btw, I don't think we need to do the centerCrop, since we are resizing to that image size already